### PR TITLE
JAMES-3156: Removal of HybridBlobStore

### PR DESCRIPTION
**Pull request by LanKhuat**

### CODE REVIEW

File: src/adr/0014-blobstore-storage-policies.md [hunk 1]
```diff
@@ -35,8 +35,6 @@ The proposed policies include:
  - LowCostStoragePolicy: The blob is expected to be saved in low cost storage. Access is expected to be unfrequent.
  - PerformantStoragePolicy: The blob is expected to be saved in performant storage. Access is expected to be frequent.
 
-An HybridBlobStore will replace current UnionBlobStore and will allow to choose between Cassandra and ObjectStorage implementations depending on the policies.
```
Comment (Arsnael) [inline src/adr/0014-blobstore-storage-policies.md:38]: I'm not sure we should make that disappear from an ADR previously written, but instead create a new ADR that supersedes this one? @chibenwa

File: upgrade-instructions.md [hunk 2]
```diff
@@ -25,7 +25,6 @@ Change list:
  - [JAMES-2703 Post 3.4.0 release removals](#james-2703-post-340-release-removals)
  - [Health checks routes return code changes](#health-checks-routes-return-code-changes)
  - [User mailboxes reIndexing endpoint change](#user-mailboxes-reindexing-endpoint-change)
- - [Hybrid blobStore replaces Union blobStore](#hybrid-blobstore-replaces-union-blobstore)
```
Comment (Arsnael) [inline upgrade-instructions.md:28]: I don't think this should be deleted... But add a new entry to say we remove it for next version. 

I think though we forgot to modify this when we released 3.5? The unreleased section is still aboutchanges between james 3.4 and 3.5

File: src/adr/0014-blobstore-storage-policies.md (hunk 1 above)
Comment (chibenwa) [reply src/adr/0014-blobstore-storage-policies.md:38]: Yes don't alter previously approuved ADRs. Removal is documented in 0025

File: upgrade-instructions.md (hunk 2 above)
Comment (chibenwa) [reply upgrade-instructions.md:28]: > I think though we forgot to modify this when we released 3.5? The unreleased section is still aboutchanges between james 3.4 and 3.5

Release is not yet official.

**Comment by Arsnael** [comment]
```
[fe11544c999999d504de786c5cb24cd6438f3bc2] [ERROR] Failures: 
[fe11544c999999d504de786c5cb24cd6438f3bc2] [ERROR]   RabbitMQEventDeadLettersIntegrationTest.multipleFailedEventShouldBeStoredInDeadLetter:351 1 expectation failed.
[fe11544c999999d504de786c5cb24cd6438f3bc2] JSON path . doesn't match.
[fe11544c999999d504de786c5cb24cd6438f3bc2] Expected: a collection with size <2>
[fe11544c999999d504de786c5cb24cd6438f3bc2]   Actual: [5cd4a0a0-664f-4016-948c-60b464284887]
```
test this please

**Comment by LanKhuat** [comment]
rebased & added JIRA ticket number.

File: upgrade-instructions.md [hunk 3]
```diff
@@ -209,24 +208,6 @@ blob.export.linshare.technical.account.password
 ```
 
 -The legacy property `blob.export.linshare.token` will not used anymore, you can remove it.
-### Hybrid blobStore replaces Union blobStore
```
Comment (rouazana) [inline upgrade-instructions.md:212]: that's strange to remove a note from the upgrade instructions... maybe we can just add that it has been removed later?

**Comment by chibenwa** [comment]
See https://github.com/linagora/james-project/pull/3339

**Pull request closed without merging**